### PR TITLE
feat: tts purchase page

### PR DESCRIPTION
**Pull request by cswills**

### CODE REVIEW

File: lib/models/purchases.dart [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      storeState = StoreState.notAvailable;
+      notifyListeners();
+      return;
+    }
+    const ids = <String>{
+      cloudTtsSubscription,
+    };
+    final response = await iapConnection.queryProductDetails(ids);
+    for (var element in response.notFoundIDs) {
+      debugPrint('Purchase $element not found');
```
Comment (kevmo314) [inline lib/models/purchases.dart:57]: Log this to Crashlytics instead so we get a remote trace.

File: lib/screens/settings/tts/cloud_tts_purchases.dart [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import 'package:rtchat/models/purchases.dart';
+import 'package:rtchat/models/user.dart';
+import 'package:rtchat/product_list.dart';
+
+class CloudTtsPurchasesScreen extends StatelessWidget {
+  const CloudTtsPurchasesScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var purchases = context.watch<Purchases>();
```
Comment (kevmo314) [inline lib/screens/settings/tts/cloud_tts_purchases.dart:14]: Could this use `FutureBuilder` instead on `Purchases.loadPurchases()`?

File: lib/screens/settings/tts/cloud_tts_purchases.dart [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    var userModel = context.watch<UserModel>();
```
Comment (kevmo314) [inline lib/screens/settings/tts/cloud_tts_purchases.dart:15]: Use `Consumer<UserModel>(...)` instead

File: lib/main.dart [hunk 4]
```diff
@@ -237,6 +238,12 @@ class _AppState extends State<App> {
                   .setString('stream_preview', jsonEncode(model.toJson()));
             });
         }),
+        ChangeNotifierProvider<Purchases>(
+          create: (context) => Purchases(
+            context.read<TtsModel>(),
+          ),
+          lazy: false,
```
[the remark this answers is not in the document]
Comment (cswills) [reply lib/main.dart:245]: Yes, when the app starts it will apply any in-app purchases such as cloud tts.

**Review by kevmo314 — APPROVED** [review]

**Review by SputNikPlop — APPROVED** [review]
This is working on iOS 16.2 iPhone 14

**Pull request merged**